src/hotspot/share/runtime/globals.hpp

```diff
@@ -543,7 +543,7 @@ const intx ObjectAlignmentInBytes = 8;
           "compression. Otherwise the level must be between 1 and 9.")      \
           range(0, 9)                                                       \
                                                                             \
-  product(ccstr, NativeMemoryTracking, "off",                               \
+  product(ccstr, NativeMemoryTracking, DEBUG_ONLY("summary") NOT_DEBUG("off"), \
```

Comment:
Hi Thomas,

I'm a bit concerned about always enabling part of NMT in debug as that affects the bulk of our hotspot testing. What is the potential overhead from enabling NMT this way? We would need to run this patch through our CI to ensure it doesn't introduce any issues.

Cheers,
David

Reply:
Hi David,

I understand your concern. I have had this patch in our internal tests enabled for several months now, but am fine with you guys doing more tests.

My arguments for this:

1) using NMT in debug builds in place of the old hard wired fences should be less overhead, not more. NMT is quite frugal. NMT=summary does:
  - increase the counter per NMT category (basically atomic-incing a number in a static array)
  - add and populate a malloc header and a footer
That's it. We did the same thing before. The old code atomically increased two global counters (`inc_stat_counter`), then added a header and a footer. Only the old header/footer had been much larger than what NMT does. So we use less memory, not more, in debug.

2) Another argument for this is that NMT is important. It gets used by support, typically in stressful situations. I rather have it thoroughly tested as part of our internal tests. I am not sure that we have ironed out every single NMT issue. NMT feels very stable to me, but there may be lingering issues. But then we should fix them.

All that said, I'd be happy for you to give this patch a spin in your CI. I can hold off the push for a bit.

Cheers, Thomas

Reply:
Thanks for the additional info Thomas. I will run it through our CI and report back.

Reply:
I'm seeing a timeout with langtools tier1 test `tools/javac/failover/CheckAttributedTree.java` on macosx-aarch64. It timed out after 9m. I'll try a few re-runs.

Reply:
10 re-runs all passed fine - maximum time 3m01s. A re-run on the same machine as previously timed-out was only 47s. So I think we are generally good to go here. Thanks.

Reply:
Thanks for doing this David.